mmdeploy/codebase/mmdet/deploy/object_detection_model.py

```diff
@@ -604,6 +604,58 @@ class labels of shape [N, num_det].
         return dets, labels.to(torch.int32)
 
 
+@__BACKEND_MODEL.register_module('vacc_det')
+class VACCDetModel(End2EndModel):
+
+    def __init__(self, backend: Backend, backend_files: Sequence[str],
+                 device: str, model_cfg: Union[str, Config],
+                 deploy_cfg: Union[str, Config], **kwargs):
+        assert backend == Backend.VACC, f'only supported vacc, but give \
+            {backend.value}'
+
+        super(VACCDetModel, self).__init__(backend, backend_files, device,
+                                           deploy_cfg, **kwargs)
+        # load cfg if necessary
+        model_cfg = load_config(model_cfg)[0]
+        self.model_cfg = model_cfg
+
+    def forward(self,
+                inputs: torch.Tensor,
+                data_samples: Optional[List[BaseDataElement]] = None,
+                mode: str = 'predict',
```

Comment:
Is `mode` necessary?

Reply:
not sure. but i notice that other model type use `mode` such as `rknn` `End2EndModel`. here `mode` can delete?

Reply:
You can add an assertion like:
```python
        assert mode == 'predict', \
            'Backend model only support mode==predict,' f' but get {mode}'
```

Reply:
there have been an assertion now
```python
assert mode == 'predict', 'Deploy model only allow mode=="predict".'
```